### PR TITLE
feat(webapp): gelatiere — a persistent resident-advisor unit that suggests skills and use cases

### PR DESCRIPTION
**Pull request by trieloff**
## Gelatiere — a resident advisor scoop that suggests skills and use cases

The gelatiere is SLICC's resident advisor: a persistent work unit that no cone owns, which reviews how the user works — every night, and after a chat session ends — and comes back with a handful of grounded suggestions: skills to install from `www.sliccy.com/skills/catalog.json` / `ai-ecoverse/skills`, use cases from the man-page sitemap, habits to change. Suggestions render as cards in the **suggestions sprinkle** (`/shared/sprinkles/suggestions/suggestions.shtml`, Install / Try it / Not now) and every other cone gets a lick.

Design write-up: `docs/gelatiere.md`. Behind the **`memory-v2`** feature flag (off by default) — the same flag that gates session search (#3034) and scoop pre-compaction snapshots (#3032); the earlier standalone `gelatiere` flag is gone.

### Shape

- **A scoop under a synthetic owner, never user-promptable.** The gelatiere is a **child** unit whose `parentJid` is the synthetic `system:gelatiere` — read-only transcript, no composer, shown in the tab strip after the cones. Every consumer of a dangling ownership edge already falls back to the default root, and cascades only follow real parents, so no cone teardown can take it down. Idle compaction (`ScoopContext`) applies to it whatever the `compact-on-idle` flag says. (The first draft made it a root cone; review feedback moved it to a scoop so the user cannot prompt it.)
- **Passes run inside its own conversation**, on licks: the worker-side crontask `gelatiere-nightly` (registered by `gelatiere init`, cron from `/shared/GELATIERE.md`), the page's `session-settled` lick after a freeze (`ui/wc/wc-gelatiere.ts`, interval-gated), or `gelatiere run`. Its charter is the record's `systemPromptAppend`; the recipe is `/shared/GELATIERE.md`, read per pass.
- **The unit never writes the store.** It writes candidates JSON and runs `gelatiere suggest <file> && gelatiere deliver`. `suggest` validates and merges (ids slugged, known ids never replaced, dismissals kept); `deliver` licks every other root cone through the seam the host publishes on `globalThis.__slicc_gelatiere` (`shell/` may not import `scoops/`).
- **The lick body carries a skill pointer, not a hint.** `data.skill` names `/workspace/skills/gelatiere/SKILL.md`; a cone reads it and posts the card stream.
- **Friendly rendering.** Gelatiere licks wear the `ice-cream-cone` icon and render a one-line headline plus the suggestion titles instead of the raw JSON; the welcome stream drops the "you're all set" and "N suggestions, newest first" copy, and the recipe requires `evidence` written as one friendly sentence to the user ("You told the welcome wizard…"), never a field dump.
- The suggestion stream is its own rail sprinkle (split from the onboarding-only welcome sprinkle, see Track C below); all three card clicks settle the store page-side, install/try continue to the cone.
- Shell: `gelatiere init / run / suggest / deliver / list / dismiss / status`. The frontmatter parser is shared with `MEMORY.md` (`base/instruction-frontmatter.ts`).

### Bundle budget

`workerEagerCeilingKb` 4292 → 4298 across rebases (documented in `first-load-budget.json`; main's own raises for #3030/#3035/#2857 stacked underneath). NOT a hoist: the delta gate measures **+3.7 kB** vs merge-base at the Track C tip (allowance 5; +1.4 kB before Track C) and every lazy lever was taken first — the unit and store modules (with the bundled `GELATIERE.md`) load via dynamic import from a fire-and-forget host hook, the command shim is 763 source bytes with lazy verb dispatch, and `base/gelatiere-constants.ts` exists precisely so `scoop-context.ts` can classify the unit without hoisting the store. What remains eager is the sync unit classification, command registration, and host wiring — which is the feature. macOS measures the worker at exactly 4296 kB and Linux CI runs ~1 kB above macOS (main: 4293 kB on Linux), so 4298 buys ~1 kB of CI headroom, not room to creep. The total-JS backstop goes 25.2 → 25.4 MB: main is 25.18 MB on CI, this branch 25.31 MB, and the ~130 kB is all lazy chunks (GELATIERE.md, DREAMING.md, the wiki skill and CLI, the memory command, dreaming and memory-health modules), none on the first-load path.

### Live exercise (local harness, Bedrock CAMP / Sonnet 4.6, three cones doing real work)

- Boot created the unit and nightly crontask; the primary wrote and ran a Node script, a `Bakery` cone built and smoke-tested a static site, a `Research` cone wrote a notes file.
- "New chat" on Bakery → session-settled lick → the gelatiere ran a 14-step pass in about a minute, produced 3 grounded suggestions (one is the bakery build–serve–screenshot loop as a workflow), wrote its notes file, replied in one line.
- The Research cone answered its lick by reading the skill, replying in one sentence and posting the cards. A forced re-delivery to each idle cone was answered at once. An orchestrator-level question about licks landing on busy/clearing cones is noted in the doc.
- Pointing `nightly` two minutes ahead and re-running `init` fired the worker-side cron into the unit on the minute; it judged nothing had changed and said so.

### Not in this PR

- Compact-on-idle on the unit was not observed live (needs an idle window on a large context; the gate is unit-tested).
- Hosted origin and extension float were not exercised.
- `crontask list` in the CLI float lists only REST tasks, so the worker-side nightly is only visible through `gelatiere status`.
- The dip card-margin CSS bug spotted during review is general to dips and gets its own PR.

### Rail persistence + card lifecycle (added after review)

- **The suggestion stream lives in the right rail** (Memory v2 only) as the `suggestions` sprinkle wearing the lucide `ice-cream-cone` icon. `sprinkle-discovery` un-hides it only when the flag is on; `welcome` stays onboarding-only and inline-locked regardless of the flag. (Earlier revisions kept the stream inside `welcome.shtml`; the split landed with Track C.)
- **Three card states**: open (buttons), **taken** (`takenAt` — Install/Try it stamp it page-side, card swaps to an in-place check + note and later folds into a "Done" ledger under the open cards), **dismissed** (`dismissedAt` — card removed). Eviction past the 40-entry cap: dismissed first, then taken, then oldest open.
- **Two demo-flushed fixes**: settlements are serialized on a module-level chain (two same-tick card clicks used to clobber each other's read-modify-write), and the panel path now consults the welcome lick interceptor (`wireWcSprinkles.interceptWelcomeLick`) — panel-hosted welcome used to leak every card click to the cone as an unhandled lick.

### Review findings addressed

- P1: flag-off boot now drops a persisted nightly crontask (`haltGelatiere`) so no billable passes continue after the user turns Memory v2 off.
- P2: `gelatiere deliver --scoop` validates the target against the roster before stamping `lastDeliveredAt`.
- P2: settlement writes serialized (above).
- Major: suggestion `url` is http(s)-only at the store boundary (it renders as an `href`; everything else renders as text), with a render-side belt in `welcome.shtml`.
- Bundle backstop lifted 25.2 → 25.4 MB (precedent: 689eac440); the strict per-change first-load ratchet passes at +3.7 kB (see Bundle budget above).
- Codex round on `7b85fd7bd` (ee2d5a019): `memory --cone` validated against existing cones before the seam (P1); `memory curate` defaults to the archive's own cone (P1); targeted `gelatiere deliver --scoop` no longer advances the delivery watermark (P2); `coerceSuggestion` enforces the kind contract — skill needs skill+validated install, use-case needs prompt (P2); the session-end interval gate is serialized (P2).
- Grok 4.6 second round on `ee2d5a019` (b50114842): curator and dreamer of one cone can no longer run at once — `AgentSpawnOptions.exclusiveWith` + `MemoryPassInstructions.rivalsFor` (P1); P7 lying-memory check and `memory status --check` are per cone (P2); `memory dream --all` runs cones sequentially so wiki writes don't race (P2); gelatiere fetches carry a 15 s timeout (P2). Its `loadJS`-typo finding was a false positive (`loadJS` is MiniSearch's plain-object loader); the misleading local annotation is fixed and a cache-hit test added.

### Track C — in this PR, gating merge

Scope agreed after the review cycle: this PR does not ship until these land here.

**Sharp gaps first (small slices):**

- [x] **P0c — archive-at-rest redaction**: `frozen-archive-writer` redacts credentials before an archive is persisted.
- [x] **Scoop-archive search indexing**: `session search` indexes `/scoops/<folder>/sessions/` (the P2 snapshots), not just `/sessions/`.

**Memory surfaces:**

- [x] **`memory` shell command** over the curator seam — inspect/manage durable memory from any cone.
- [x] **Gelatiere "dreaming" phase** — nightly memory-refactoring pass (consolidate/rewrite, not just suggest).
- [x] **`/shared/wiki/`** per the llm-wiki skill (`ai-ecoverse/skills` → `skills/llm-wiki/`), plus the Tier-0 `wiki.jsh` fix.
- [x] **P6 — provenance/supersession**: memory entries carry origin and what they replace.
- [x] **P7 — enforced verified maintenance**: memory upkeep checked by the runtime, not requested of the model.

**Deferred review findings:**

- [x] **Split the gelatiere stream from onboarding on follower/extension surfaces** (Grok #4): the stream is now `/shared/sprinkles/suggestions/suggestions.shtml`, `welcome.shtml` is onboarding-only again, followers drop stream dips (leader-local state) and retract a stale "Set up SLICC" card once a stream dip proves onboarding finished, and `suggestions-sprinkle-split.test.ts` pins the split.
- [ ] *(Optional, cosmetic)* per-`lickPart` `gelatiereAction` in the collated lick card (review #7).

### CODE REVIEW

File: packages/webapp/src/kernel/host.ts [hunk 1]
```diff
@@ -1049,6 +1050,45 @@ export function shouldStartLickWsBridge(adapter: CapabilityAdapterId): boolean {
...
+  orchestrator: OrchestratorType,
+  lickManager: LickManager,
+  sharedFs: VirtualFS | null,
+  log: KernelHostLogger
+): void {
+  void import('../scoops/gelatiere-unit.js')
+    .then(async (unit) => {
+      const seam = unit.createGelatiereSeam(orchestrator, lickManager);
+      unit.publishGelatiereSeam(seam);
+      if (!sharedFs || !isFeatureEnabled('memory-v2')) return;
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/src/kernel/host.ts]: Fixed in 9f3b32ea3. The flag-off path now calls the new `haltGelatiere(seam)`, which drops the persisted `gelatiere-nightly` crontask via a new seam method `dropNightly()` — so a nightly scheduled while memory-v2 was on stops firing after the flag goes off. Boot order is safe: `publishGelatiere` runs at step 8b, after `lickManager.init()` (step 7) has loaded persisted crontasks, so the halt sees them. The unit itself stays as a frozen transcript (it acts only on licks); flipping the flag back on reschedules the nightly through `bootGelatiere`. Covered by a new test in `gelatiere-unit.test.ts`.

File: packages/webapp/src/shell/supplemental-commands/gelatiere/run.ts [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+  const added = store.suggestionsSince(all, state.lastDeliveredAt);
+  if (added.length === 0 && !parsed.bools.has('--force')) {
+    return ok('Nothing new since the last delivery; no lick sent (use --force to resend).\n');
+  }
+  const explicit = parsed.values.get('--scoop');
+  const targets = explicit ? [explicit] : host.roots().map((r) => r.folder);
+  if (targets.length === 0) return fail('no cone is running to deliver to');
+  const body = store.buildGelatiereLickBody(added, open);
+  for (const target of targets) host.lick(target, body);
+  await store.writeGelatiereState(fs, { ...state, lastDeliveredAt: new Date().toISOString() });
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/src/shell/supplemental-commands/gelatiere/run.ts]: Fixed in 9f3b32ea3. An explicit `--scoop` target must now resolve against `host.roots()` (folder, name, or jid — the aliases the lick router accepts) before anything is sent; an unknown target fails with the roster listed and does NOT stamp `lastDeliveredAt`, so the suggestions stay "new" for the next delivery. Help text updated to "One cone (folder or jid)". Test added asserting the ledger is untouched on the failure path.

File: packages/webapp/src/ui/boot/setup-welcome-flow.ts [hunk 3]
```diff
@@ -310,8 +323,46 @@ const WELCOME_BRANCHES: Record<
...
+  return gelatiereModule;
+}
+
+function settleGelatiereSuggestion(body: WelcomeBranchBody, deps: WelcomeBranchDeps): void {
+  const id = (body?.data as WelcomeGelatiereCardData | undefined)?.id;
+  if (typeof id !== 'string' || !id || !deps.vfs) return;
+  const vfs = deps.vfs;
+  void loadGelatiereModule()
+    .then(({ dismissGelatiereSuggestion }) => dismissGelatiereSuggestion(vfs, id))
+    .catch((err) => deps.log.warn('Failed to settle gelatiere suggestion', err));
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/src/ui/boot/setup-welcome-flow.ts:376]: Fixed in d931f8380 — and confirmed real: a live CDP demo hit exactly this (Install + Not now in the same tick left the store at "2 open, 0 taken"; the second read-modify-write clobbered the first). Settlements now run on a module-level promise chain in `gelatiere-store.ts` (`settleSuggestion` serializes both `dismissGelatiereSuggestion` and the new `takeGelatiereSuggestion`), so concurrent card clicks can't lose writes. Concurrency test added (`serializes concurrent settlements so neither write clobbers the other`).

File: packages/webapp/src/base/gelatiere-store.ts [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+  const stamp = createdAt ?? optionalText(entry.createdAt, 40) ?? '';
+  return {
+    id,
+    kind: kind as GelatiereSuggestionKind,
+    title,
+    body,
+    ...(entry.skill !== undefined ? { skill: optionalText(entry.skill, 120) } : {}),
+    ...(entry.install !== undefined ? { install: optionalText(entry.install, 300) } : {}),
+    ...(entry.prompt !== undefined ? { prompt: optionalText(entry.prompt, 1_000) } : {}),
+    ...(entry.url !== undefined ? { url: optionalText(entry.url, 500) } : {}),
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/src/base/gelatiere-store.ts]: Fixed in 9f3b32ea3, at the boundary you suggested: `coerceSuggestion` now passes `url` through a `httpUrl()` guard (parses with `new URL`, keeps only `http:`/`https:`, else `undefined`), which also covers the lick-body consumers and previously-stored bad URLs (the read path re-coerces). Added a render-side belt in `welcome.shtml` (`/^https?:\/\//i` before assigning `href`). Test covers `javascript:`, `data:`, `vbscript:`, and non-URL strings.

**Comment by trieloff** [comment]
## Live demo (local harness, Bedrock / Sonnet 4.6)

Suggestion stream in the right rail — design before/after (nested roundrect cards → flat entries with the what/why copy contract):

| Before | After |
| --- | --- |
| before | after |

Boot, with the gelatiere's read-only tab in the strip:

boot

Card lifecycle — "Try it" just clicked (buttons swap for a quiet in-place confirmation, lick continues to the cone):

taken

After a re-render: taken entries fold into the Done ledger, and in the chat the cone is genuinely acting on the forwarded lick (it wrote and ran the workflow file the suggestion proposed):

done

**Comment by trieloff** [comment]
Follow-up to the redesign shots above (`75abadaf8`): actions row is now right-aligned — quiet "Not now" beside the primary pill — and entry padding got a touch more air.

actions right-aligned

**Review by trieloff — COMMENTED** [review]
Reviewed at 75abadaf8 (the rebased branch, including the Memory v2 follow-ups). CI is green apart from e2e still running. I also drove the earlier tip live on a local harness with Bedrock: session-end lick → pass → grounded suggestions with second-person evidence, cards rendering without the wizard shell, the gelatiere lick card with its own icon. The Codex/bot findings from efcb0cea2 are all addressed in 9f3b32ea3 and d931f8380 as described. What follows is what is left.

### Should fix before merge

**1. The session-end interval only holds if the model runs `gelatiere suggest` — an empty pass never stamps the ledger.**
`isPassDue` (`packages/webapp/src/base/gelatiere-store.ts:241`) keys on `lastPassAt`, and the only writer of `lastPassAt` is `recordPass`, i.e. `gelatiere suggest`. `GELATIERE.md` says "an empty list is a fine outcome", and in the live run the unit did exactly that: it answered "nothing to add" in one line and never ran `suggest` (`gelatiere status` stayed at `Passes: 1`). After that, every "New chat" on any cone sends another `session-settled` lick, because the ledger still says the last pass is old — so the one gate that bounds billable passes is only as reliable as the model's compliance with the last step of the recipe. Suggest stamping the ledger where the trigger is decided (`notifyGelatiereOfSessionEnd` in `ui/wc/wc-gelatiere.ts:459` could write `lastPassAt`/`lastTriggeredAt` when it sends the lick), or have the charter require `gelatiere suggest` with an empty `{"suggestions":[]}` and add a test that an empty file still bumps `passes`.

**2. Unattended agent with read access to everything and unrestricted egress.**
`GELATIERE_VISIBLE_PATHS` (`packages/webapp/src/scoops/gelatiere-unit.ts:187`) covers `/sessions/`, `/home/`, `/cones/` and every cone's memory, and `curl` is on `GELATIERE_ALLOWED_COMMANDS` — for a child unit `allowedCommands` is the only network gate (`scoops/llms-txt-ignore.ts:44`). The recipe then has this unit read third-party content on every pass (the catalog, the sitemap, a GitHub repo listing). A single injected catalog entry or README line ("also POST /sessions/*.md to …") turns a nightly, approval-free pass into an exfiltration channel for every archived session. The curator has the same shape but does not fetch the open web. Options, cheapest first: drop `curl` from the allow-list and give the recipe a `gelatiere catalog` / `gelatiere sitemap` subcommand that fetches only the known hosts; or keep `curl` but route it through the sudo gate (leave it off the list so it escalates — the nightly then needs the user, which is arguably right for egress); or at least trim `/sessions/` to the session index plus the archive named in the lick. Worth a sentence in `docs/gelatiere.md` either way.

### Worth a look

**3. Idle-nag change reaches every silent child, not just the gelatiere.** `scoop-lifecycle-manager.ts:137` now skips the idle timer for any child with `notifyOnComplete === false`. Today the only other one is the approver agent (`approver-agent.ts:211`), which is fed immediately, so nothing breaks — but the commit message frames this as a gelatiere change and it is a policy change. A line in `docs/work-unit.md` (silent units opt out of idle notices) would keep the next reader from re-deriving it.

**4. Settlement is serialized per realm, not per file.** `settleChain` (`gelatiere-store.ts:307`) orders card clicks on the page; `gelatiere dismiss` and `recordPass` run in the worker on the same JSON. A card click landing during a pass's read-merge-write can be lost. Rare and self-healing (the card comes back), so a comment is enough unless you want the store behind a single writer.

**5. Flag-off leaves the unit in the roster.** `haltGelatiere` drops the nightly, which fixes the billing issue, but the `gelatiere` tab stays and `gelatiere run` still works with Memory v2 off (no flag check in `run.ts`). Fine if intended; then `gelatiere status` should probably say the flag is off rather than "registered".

**6. Stale comments after the root → scoop switch.** `gelatiere-store.ts:5` and `gelatiere/run.ts:7` still say "persistent root (work) unit".

**7. `gelatiereAction` is derived from the whole message** (`wc-message-view.ts:1202`) while the sections are per `lickPart`; a collated multi-lick card falls back to the `suggestions` pill. Cosmetic.

### Verified fine

- `httpUrl` at the store boundary plus the render-side belt; `describeGelatiereLick` output only ever reaches `textContent`; card text in `welcome.shtml` goes through `textContent` too.
- `ensureGelatiereUnit` refusing a taken folder and `init --reset` deleting crontasks before units — this is exactly the sequence that wedged the harness before the fix.
- `deliver --scoop` validating against `roots()` before stamping `lastDeliveredAt`.
- The bundle gates: the unit and store modules are lazy on both the host and the command path, and first-load passed on my run.

Not exercised: compact-on-idle on the unit, the hosted origin, the extension float — same list as the PR body.

**Comment by trieloff** [comment]
All seven findings from the 19:03 review addressed in 860c7b51a (Codex's independent pass on the same diff overlapped on two of them; its third — hosted-leader card settlement — is folded in too).

**1. Empty-pass re-fire** — fixed at the trigger, as suggested. `notifyGelatiereOfSessionEnd` stamps a new `lastTriggeredAt` the moment it sends the lick, and `isPassDue` gates on the newest of `lastPassAt`/`lastTriggeredAt`. Test reproduces the exact live scenario: lick sent, no `gelatiere suggest`, next "New chat" inside the interval does not re-lick.

**2. Egress** — took the cheapest option and then some. `curl` is off `GELATIERE_ALLOWED_COMMANDS` (so any raw fetch now escalates through the sudo gate, which is right for egress), and the recipe's three known fetches became `gelatiere catalog` / `gelatiere commands` / `gelatiere man <cmd>`, pinned to `www.sliccy.com` with byte caps and a slug check on the man name. `GELATIERE.md` now tells the unit "you have no curl and need none". The unit test asserts curl/wget/nc/ssh stay off the list, and `docs/gelatiere.md` carries the rationale sentence. Related belt: `install` is now validated at the store boundary as a plain `upskill` invocation (bare tokens, no shell metacharacters) — it is the one field a cone executes verbatim, and the pass reads third-party content (also Codex's P1).

**3. Idle-nag policy** — documented in `docs/work-unit.md`: silent units (`notifyOnComplete: false`) opt out of idle notices along with completion reports.

**4. Cross-realm settle race** — comment on `settleChain` now states the per-realm limit, why the loss is rare and self-healing, and when a single writer would become worth it.

**5. Flag-off roster** — `gelatiere status` now leads with `Memory v2: OFF — the nightly is unscheduled and session ends do not trigger passes` when the flag is off, so "registered" doesn't read as "active". `run` stays usable by design (manual poke).

**6. Stale comments** — both "persistent root (work) unit" occurrences now say "persistent scoop".

**7. Per-message `gelatiereAction`** — acknowledged, left as is: collated multi-lick cards fall back to the `suggestions` pill, which is cosmetically wrong but never misleading, and the collation path is shared with every other lick kind. Happy to take it in Track C if it bothers.

Also from Codex: cherry/hosted-leader floats skip `wireWcWelcome`, so gelatiere card clicks forwarded to the cone unsettled there. A small fallback interceptor (`wc-gelatiere-fallback.ts`) now settles dismiss/install/try in those floats with the same consume/forward semantics; the store module stays lazy.

Full pass locally: lint, typecheck, 21,383 tests, webapp coverage gate, debt gate, both builds — all green.

**Comment by trieloff** [comment]
## Grok 4.6 review (sourced independently) — findings and disposition

Four findings on the merge-base diff; two were already fixed by 860c7b51a this morning, one is hardened in 74a57bf57, one is deferred with rationale.

**[P1] Cone executes lick-supplied `install`/`prompt`** — hardened in 74a57bf57. The store-boundary upskill validation (860c7b51a) closed the injected-suggestion path, but Grok is right that a forged `gelatiere-install` lick (a full-trust follower, another sprinkle) carries its own `install` and the skill said "run verbatim". The `gelatiere` skill now has the cone resolve the suggestion by `id` from the store (`gelatiere list --all --json`) and act on the stored, validated fields only — unknown id means stop. Page-side rewrite of the forwarded body would need async work in a sync interceptor; store-resolution at the cone gets the same guarantee without restructuring the lick pipeline.

**[P2] Claim the interval at trigger time** — already fixed in 860c7b51a (`lastTriggeredAt`, stamped when the lick is sent; `isPassDue` gates on the newest stamp). Same finding as the 19:03 review's #1.

**[P2] `recordPass` outside the settle chain** — same as the 19:03 review's #4 and Codex's race finding; documented on `settleChain` in 860c7b51a per the reviewer's "a comment is enough". The loss window is one microtask inside a minutes-long pass and self-heals (the card re-renders); a cross-realm single writer stays on the shelf until that stops being true.

**[P2] `welcome.shtml` behaves like first-run onboarding on follower/extension surfaces** — real cross-runtime parity gap, deferred to Track C. Follower-rendered sprinkles hard-code `exists()` false so `bootByMarker` restarts the wizard, and the extension panel swaps welcome images for a "Set up SLICC" handoff card. The PR body already scopes follower/extension surfaces out ("not exercised"); splitting the suggestion stream from the onboarding sprinkle (or teaching the follower path about the welcomed marker) is a surface change that deserves its own slice rather than a rider here.

File: packages/webapp/src/shell/supplemental-commands/memory/run.ts [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+  if (all) {
+    // Every cone that actually has a memory file — a cone that never
+    // accumulated memory has nothing to consolidate.
+    folders = [];
+    for (const folder of await listConeFolders(fs)) {
+      if ((await readMemoryFile(fs, memoryPathFor(folder))) !== null) folders.push(folder);
+    }
+    if (folders.length === 0) return fail('no cone has a memory file yet — nothing to dream about');
+  } else {
+    folders = [cone ?? PRIMARY_CONE_FOLDER];
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/src/shell/supplemental-commands/memory/run.ts]: Fixed in ee2d5a019. `memory curate|dream|show --cone` now resolves the value against the cones that actually exist (`cone`, or a `/cones/<folder>` directory) before it crosses the seam — `resolveConeFolder` in `memory/run.ts`. `../../shared`, a typo or a dropped cone fails with `unknown cone "<x>" (cones: …)` and the seam is never called. Tests: `memory-command.test.ts` ("rejects a --cone that is not an existing cone folder before crossing the seam", for both `curate` and `dream`).

File: packages/webapp/src/shell/supplemental-commands/memory/run.ts [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+    if (!entry) return fail('no archived sessions to curate — freeze a chat first');
+  } else {
+    entry = index.find((candidate) => candidate.filename === wanted);
+    if (!entry) return fail(`no archive named "${wanted}" in /sessions/index.json`);
+  }
+  const folder = parsed.values.get('--cone');
+  const result = await host.curate({
+    sessionArchivePath: `/sessions/${entry.filename}`,
+    sessionCount: index.length,
+    ...(folder && folder !== PRIMARY_CONE_FOLDER ? { cone: { folder } } : {}),
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/src/shell/supplemental-commands/memory/run.ts]: Fixed in ee2d5a019. `memory curate` without `--cone` now targets the cone the archive was frozen from (the index entry's `cone`, absent on legacy entries = the primary). An explicit `--cone` is validated against existing cones; an archive whose cone no longer exists fails with a message asking for `--cone <folder>` rather than silently landing in `/workspace/CLAUDE.md`. Tests: "defaults the target to the archive's own cone" and "refuses an archive whose cone is gone unless --cone redirects it".

File: packages/webapp/src/shell/supplemental-commands/gelatiere/run.ts [hunk 7]
```diff
@@ -0,0 +1,357 @@
...
+    r.folder === explicit || r.name === explicit || r.jid === explicit;
+  if (explicit && !roster.some(resolves)) {
+    const known = roster.map((r) => r.folder).join(', ') || 'none running';
+    return fail(`unknown delivery target "${explicit}" (cones: ${known})`);
+  }
+  const targets = explicit ? [explicit] : roster.map((r) => r.folder);
+  if (targets.length === 0) return fail('no cone is running to deliver to');
+  const body = store.buildGelatiereLickBody(added, open);
+  for (const target of targets) host.lick(target, body);
+  await store.writeGelatiereState(fs, { ...state, lastDeliveredAt: new Date().toISOString() });
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/src/shell/supplemental-commands/gelatiere/run.ts]: Fixed in ee2d5a019. A targeted `gelatiere deliver --scoop <target>` no longer advances `lastDeliveredAt` — the watermark records what every cone has been told, so only a broadcast stamps it. The output says so (`… (targeted; the delivery watermark is unchanged)`), and `--help` documents it. Per-cone delivery state was considered and rejected as over-engineering: the recipe only ever runs the untargeted form, and a repeat lick to the one cone that was targeted is the cheap side of the trade. Test: "a targeted deliver leaves the watermark alone so a later broadcast still reaches the rest".

File: packages/webapp/src/base/gelatiere-store.ts [hunk 8]
```diff
@@ -0,0 +1,659 @@
...
+ * wins for fresh output; `null` means "reading the store back", where the
+ * stored stamp is the truth.
+ */
+function coerceSuggestion(raw: unknown, createdAt: string | null): GelatiereSuggestion | null {
+  if (!raw || typeof raw !== 'object') return null;
+  const entry = raw as RawSuggestion;
+  const kind = typeof entry.kind === 'string' ? entry.kind : '';
+  const title = optionalText(entry.title, 200);
+  const body = optionalText(entry.body, 1_000);
+  if (!SUGGESTION_KINDS.has(kind) || !title || !body) return null;
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/src/base/gelatiere-store.ts:444]: Fixed in ee2d5a019. `coerceSuggestion` now enforces the kind contract `GELATIERE.md` documents: a `skill` needs `skill` and a validated `install` (missing, or rejected by `upskillInstall`, drops the candidate), and a `use-case` needs `prompt`. `tip` stays body-only. So every card the stream renders an Install / Try it button for has the field behind that button. The install-validation test was reshaped accordingly (rejected commands now drop the suggestion instead of surviving with `install: undefined`), plus a new "drops a skill without a usable install command and a use-case without a prompt" case.

File: packages/webapp/src/ui/wc/wc-gelatiere.ts [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    if (!unit) {
+      deps.log.debug('gelatiere unit not registered; session end not announced');
+      return false;
+    }
+    const [config, state] = await Promise.all([
+      loadGelatiereConfig(deps.vfs),
+      readGelatiereState(deps.vfs),
+    ]);
+    const now = (deps.now ?? (() => new Date()))();
+    if (!isPassDue(state, now, config.intervalHours)) {
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/src/ui/wc/wc-gelatiere.ts:80]: Fixed in ee2d5a019. `notifyGelatiereOfSessionEnd` now queues every call on a module-level promise chain, so the read → `isPassDue` → `recordGelatiereTrigger` sequence of the second caller runs after the first has stamped `lastTriggeredAt`. Test: "serializes concurrent session ends so only one passes the interval gate" (two `Promise.all`-ed session ends → one lick, results `[true, false]`).

**Comment by trieloff** [comment]
## Grok 4.6 re-review (sourced independently, second round) — findings and disposition

Reviewed at `ee2d5a019` (after the Codex round). Grok confirmed all four of its first-round findings as properly resolved, then raised one P1 and four P2s on the Track C additions. Four are fixed in b50114842; one is a false positive, with the misleading annotation that invited it fixed.

**[P1] Curator and dreamer merge the same live memory file without colliding** — fixed. Both passes snapshot a base and three-way-merge a draft onto the same file; under different fixed names (`memory-curator-<cone>` / `memory-dreamer-<cone>`) the bridge let them run at once, and the later merge discarded the earlier rewrite. `AgentSpawnOptions.exclusiveWith` now makes the bridge reject a spawn whose named rival is live, on the same `name already in use` path as a live namesake, and each pass names the other via `MemoryPassInstructions.rivalsFor`. So a "New chat" mid-dream leaves that archive pending for the boot catch-up (the existing collided path), and a nightly mid-curation skips that cone until the next night. Tests on the bridge, the curator's and the dreamer's spawn options.

**[P2] `session search` never loads the on-disk index (`loadJS` vs `loadJSON`)** — not a bug. `MiniSearch.loadJS(js: AsPlainObject, …)` is the real API for an already-parsed index object; `loadJSON` takes a string. The persisted index is `JSON.parse`d before loading, so `loadJS` is the correct call and the cache was working. What invited the finding was our local `MiniSearchCtor` annotation, which said `loadJSON` — fixed. Grok's suggested test was worth having regardless: a repeat search with an unchanged fingerprint is now pinned to read `/sessions/.search-index.json` and none of the archives.

**[P2] P7 lying-memory check is primary-cone only** — fixed. Curated archives are grouped by their `cone` and each cone's own memory file is checked, in both the scheduled runtime check (`MemoryHealthReport.cones`) and `memory status --check`. A healthy extra-cone setup with no primary memory no longer fails every boot; an extra cone whose file vanished no longer passes; archives of a cone that was dropped are not counted as a lie.

**[P2] Wiki writes from parallel dreamers are unstaged** — fixed at the trigger. `memory dream --all` runs the cones one after another even when detached, so no two dreamers touch `/shared/wiki/index.md` / `log.md` at once. Staging the wiki like the memory draft stays on the shelf: the sequential nightly removes the race, and the wiki is one shared tree rather than one file per cone.

**[P2] Gelatiere catalog/man fetch has no timeout** — fixed. `fetchSliccy` uses `AbortSignal.timeout(15_000)`; the three verbs are the unattended nightly pass's whole web surface. Tests assert the signal on every call.

Docs: `docs/gelatiere.md` and `docs/shell-reference.md` describe the exclusion and the sequential `--all`.
